extensions/kusto/src/kustoServer.ts

```diff
@@ -0,0 +1,173 @@
+			// const path = "e:\\repos\\azuredatastudio\\extensions\\kusto\\sqltoolsservice\\Windows\\2.0.0-release.15\\MicrosoftKustoServiceLayer.exe";
+			const installationComplete = Date.now();
+			let serverOptions = generateServerOptions(context.extensionContext.logPath, path);
+			let clientOptions = getClientOptions(context);
+			this.client = new SqlOpsDataClient(Constants.serviceName, serverOptions, clientOptions); // TodoKusto: Update constant
+			const processStart = Date.now();
+			const clientReadyPromise = this.client.onReady().then(() => {
+				const processEnd = Date.now();
+				statusView.text = localize('serviceStartedStatusMsg', "{0} Started", Constants.serviceName);
+				setTimeout(() => {
+					statusView.hide();
+				}, 1500);
+				vscode.commands.registerCommand('kusto.loadCompletionExtension', (params: CompletionExtensionParams) => {
+					this.client.sendRequest(CompletionExtLoadRequest.type, params);
+				});
+				Telemetry.sendTelemetryEvent('startup/LanguageClientStarted', {
+					installationTime: String(installationComplete - installationStart),
+					processStartupTime: String(processEnd - processStart),
+					totalTime: String(processEnd - installationStart),
+					beginningTimestamp: String(installationStart)
+				});
+			});
+			statusView.show();
+			statusView.text = localize('startingServiceStatusMsg', "Starting {0}", Constants.serviceName);
+			this.client.start();
+			await Promise.all([this.activateFeatures(context), clientReadyPromise]);
+			return this.client;
+		} catch (e) {
+			Telemetry.sendTelemetryEvent('ServiceInitializingFailed');
+			vscode.window.showErrorMessage(localize('failedToStartServiceErrorMsg', "Failed to start {0}", Constants.serviceName));
+			throw e;
+		}
+	}
+
+	private async download(context: AppContext): Promise<string> {
+		const rawConfig = await fs.readFile(path.join(context.extensionContext.extensionPath, 'config.json')); // TodoKusto: Update config.json to refer to the right exe
+		this.config = JSON.parse(rawConfig.toString());
+		this.config.installDirectory = path.join(__dirname, this.config.installDirectory);
+		this.config.proxy = vscode.workspace.getConfiguration('http').get('proxy');
+		this.config.strictSSL = vscode.workspace.getConfiguration('http').get('proxyStrictSSL') || true;
+
+		const serverdownloader = new ServerProvider(this.config);
+		serverdownloader.eventEmitter.onAny(generateHandleServerProviderEvent());
+		return serverdownloader.getOrDownloadServer();
+	}
+
+	private activateFeatures(context: AppContext): Promise<void> {
+		const credsStore = new CredentialStore(context.extensionContext.logPath, this.config);
+		const resourceProvider = new AzureResourceProvider(context.extensionContext.logPath, this.config);
+		this.disposables.push(credsStore);
+		this.disposables.push(resourceProvider);
+		return Promise.all([credsStore.start(), resourceProvider.start()]).then();
```

Comment:
the .then() still doesn't make any sense here

Reply:
This was removed.